staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go

```diff
@@ -227,6 +261,7 @@ func TestGracefulTerminationWithKeepListeningDuringGracefulTerminationDisabled(t
 	lifecycleSignalOrderExpected := []string{
 		string("ShutdownInitiated"),
 		string("AfterShutdownDelayDuration"),
+		string("PreShutdownHooksCompleted"),
 		string("HTTPServerStoppedListening"),
 		string("InFlightRequestsDrained"),
 	}
```

Comment:
```
genericapiserver_graceful_termination_test.go:419: Expected order of termination event signal to match, diff:   []string{
          	"ShutdownInitiated",
        + 	"PreShutdownHooksCompleted",
          	"AfterShutdownDelayDuration",
          	"InFlightRequestsDrained",
          	"HTTPServerStoppedListening",
          }
```


?

Reply:
interesting i ran `stress` on it:
```
$ stress ./server.test -test.run=TestGracefulTerminationWithKeepListeningDuringGracefulTerminationDisabled
5s: 24 runs so far, 0 failures
10s: 48 runs so far, 0 failures
...
5m5s: 1608 runs so far, 0 failures
5m10s: 1637 runs so far, 0 failures
```

@aojea how are you running the test, did you have any changes on the PR?

Reply:
It was actually the run from CI IIUC...

Reply:
oh, this failure is from the other test `TestGracefulTerminationWithKeepListeningDuringGracefulTerminationEnabled` which exercises the other mode of graceful termination that is enabled with `shutdown-send-retry-after`. 

I have yet to fix this test and `TestPreShutdownHooks/ShutdownSendRetryAfter_is_enabled`, that's why i left the PR as WIP. I will fix it and put the PR up for review.